### ISSUE TITLE
Training Intent to Code using Transformer 

### ISSUE THREAD

**Issue by JohnlNguyen** [JohnlNguyen/semantic_code_search#1]
## Metadata about the problem
* training data is unshuffled
* I used rewritten intent however if the rewritten is missing from a row I will use intent 
* typical sequence length is around 20

## Rewritten Intent to Code 6 Layers
**Test BLEU .30**

approx_vocab_size = 2**13 ~ 8k

### Hyperparameters

[Full hyperparameters set](https://pastebin.com/0PUY7D6C)

 ```json
 'batch_shuffle_size': 512,
 'batch_size': 1024,
 'hidden_size': 512,
 'learning_rate': 0.4,
 'learning_rate_decay_steps': 5000,
 'learning_rate_warmup_steps': 400,
 'length_bucket_step': 1.1,
 'max_length': 256,
 'max_relative_position': 0,
 'max_target_seq_length': 0,
 'min_length_bucket': 8,
 'num_heads': 8,
 'num_hidden_layers': 6,
 'self_attention_type': 'dot_product',
 'train_steps': 10000000,
```

Training Loss
![Screen Shot 2019-05-08 at 11 02 29 PM](https://user-images.githubusercontent.com/14943751/57430727-6fe15a80-71e5-11e9-87ea-0c7f3516ff68.png)

## Rewritten Intent to Code 2 Layers

Same hyperparameters as above however num_hidden_layers==2
![Screen Shot 2019-05-08 at 11 37 43 PM](https://user-images.githubusercontent.com/14943751/57432300-54c51980-71ea-11e9-9a73-044d10866369.png)

### Code to Rewritten Intent
![Screen Shot 2019-05-09 at 12 22 21 AM](https://user-images.githubusercontent.com/14943751/57434754-922ca580-71f0-11e9-893a-9a25ca9b3b84.png)
